### PR TITLE
Move generated code to directories with _gen suffix

### PR DESCRIPTION
**Pull request by sejr**
This PR addresses issue #602.

cc @andreeaflorescu

### CODE REVIEW

**Comment by raduweiss** [comment]
@sejr you might have noticed that you can't access the test fail error. Sorry about that! We're still working to make those logs public. Until then, the integration tests we run are exactly the ones you can run with `tools/devtool test`. See the [testing section of the getting started guide](https://github.com/firecracker-microvm/firecracker/blob/master/docs/getting-started.md#running-the-integration-test-suite) for details.

**Comment by dianpopa** [comment]
@sejr your PR does not pass the build as there are dependencies in the code where you did not use the newly updated name of the crates. For example, [this file](https://github.com/sejr/firecracker/blob/move-gen-code/cpuid/Cargo.toml) still looks for `kvm_sys`.  Additionally, after "toml" dependencies have been updated, you will also need to update the `use` scenarios: e.g. `use kvm_sys::*;` from [this file](https://github.com/sejr/firecracker/blob/move-gen-code/kvm/src/cap.rs).

**Comment by sejr** [comment]
@raduweiss When I run that I get an error: `/dev/kvm not found` any thoughts on fixing this, as I'm working locally on a Mac?

@dianpopa Oops. Fixed :)

**Comment by andreeaflorescu** [comment]
Hey, the changes look good, but can you please squash your commits? We want each individual commit on the master branch to build and pass the unit tests.

**Comment by sejr** [comment]
@acatangiu @andreeaflorescu  Commits have been squashed.

File: kvm_gen/src/x86/bindings.rs [hunk 1]
```diff
@@ -267,9 +267,9 @@ pub const KVM_S390_RESET_CLEAR: ::std::os::raw::c_uint = 2;
 pub const KVM_S390_RESET_SUBSYSTEM: ::std::os::raw::c_uint = 4;
 pub const KVM_S390_RESET_CPU_INIT: ::std::os::raw::c_uint = 8;
 pub const KVM_S390_RESET_IPL: ::std::os::raw::c_uint = 16;
-pub const KVM_SYSTEM_EVENT_SHUTDOWN: ::std::os::raw::c_uint = 1;
-pub const KVM_SYSTEM_EVENT_RESET: ::std::os::raw::c_uint = 2;
-pub const KVM_SYSTEM_EVENT_CRASH: ::std::os::raw::c_uint = 3;
+pub const kvm_genTEM_EVENT_SHUTDOWN: ::std::os::raw::c_uint = 1;
```
Comment (alexandruag) [inline kvm_gen/src/x86/bindings.rs]: It looks like these lines were affected by some sort of find/replace collateral damage.

Comment (acatangiu) [inline kvm_gen/src/x86/bindings.rs]: This doesn't look right.

Comment (sejr) [reply kvm_gen/src/x86/bindings.rs]: @alexandruag Fixed.

**Pull request merged**